packages/devtools_app/lib/src/inspector/inspector_breadcrumbs.dart

```diff
@@ -0,0 +1,169 @@
+import 'package:flutter/material.dart';
+
+import '../../devtools_app.dart';
+import 'inspector_text_styles.dart';
+
+class InspectorBreadcrumbNavigator extends StatelessWidget {
+  const InspectorBreadcrumbNavigator({
+    Key key,
+    @required this.rows,
+    @required this.onTap,
+  }) : super(key: key);
+
+  final List<InspectorTreeRow> rows;
+  final Function(InspectorTreeRow) onTap;
+
+  @override
+  Widget build(BuildContext context) {
+    if (rows.isEmpty) {
+      return const SizedBox();
+    }
+
+    final items = _getBreadcrumbs(rows);
+    final breadcrumbs = _getBreadcrumbWithChevron(items);
+    return SizedBox(
+      height: isDense() ? 24 : 28,
+      child: Padding(
+        padding: const EdgeInsets.symmetric(horizontal: 6),
+        child: Row(
+          children: breadcrumbs.map((item) {
+            if (item.isChevron) {
+              return Icon(
+                Icons.chevron_right,
+                size: defaultIconSize,
+              );
+            }
+
+            return Flexible(
+              child: _InspectorBreadcrumb(
+                data: item,
+                onTap: () => onTap(item.row),
+              ),
+            );
+          }).toList(),
+        ),
+      ),
+    );
+  }
+
+  List<_InspectorBreadcrumbData> _getBreadcrumbs(List<InspectorTreeRow> rows) {
+    final List<_InspectorBreadcrumbData> items = rows
+        .map((e) => _InspectorBreadcrumbData.wrap(e, e == rows.safeLast))
+        .toList();
+    if (items.length > 5) {
```

Comment:
Max number of breadcrumbs is 4 (excluding "[root]" and "…"), this limit is enforced on that line. If there are more breadcrumbs between root and the first visible breadcrumb, ellipsis item is displayed. It should actually be 4 instead of 5 in code. I've changed it to 4 and pulled this value out into a const.